src/foam/nanos/fs/TextSaveView.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.fs',
+  name: 'TextSaveView',
+  extends: 'foam.u2.Element',
+
+  documentation: 'view to save plain text as file',
+
+  requires: [
+    'foam.blob.BlobBlob',
+    'foam.nanos.fs.File'
+  ],
+
+  properties: [
+    'data'
+  ],
+
+  methods: [
+    function initE() {
+      this.SUPER();
+      var name = "";
+
+      this
+        .start('p')
+          .add('File Name')
+        .end()
+        .start('input')
+          .attrs({
+            name: 'fileName'
+          })
+        .end()
+        .br()
+        .br()
+        .start('p')
+          .add('Text')
+        .end()
+        .start('input')
+          .on('blur', this.onBlur)
+        .end();
+    }
+  ],
+
+  listeners: [
+    function onBlur(e) {
+      let text = e.target.value;
+      let blob = new Blob([text], {
+          type: 'text/plain'
+      });
+      let name = document.getElementsByName('fileName');
```

Comment:
This will prevent you from using two of these views on the same screen. Why not just create a property named 'fileName', then access the property value directly.

Reply:
`.on('blur', this.setName)`

`function setName(e) {`
`      this.fileName = e.target.value;`
`    }`